executor/executor.go

```diff
@@ -180,6 +180,10 @@ func (e *Executor) Run(keys state.Keys, f func() error) {
 				// and can't mark itself as executed until all [shared] are
 				// cleared (which can't be done while we hold the lock for [lt]).
 				for _, rt := range lt.readers {
+					// Don't block on ourself if we just recorded us reading
```

Comment:
As mentioned in DMs, I think there is still a weird scenario this doesn't address (assuming neither tx1 nor tx2 are executed):
```
tx1: A:read, B:write, C:read
tx2: A:read, B:write, C:read
```